okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -1080,16 +1081,23 @@ public void settings(boolean clearPrevious, Settings settings) {
         if (OkHttpSettingsUtil.isSet(settings, OkHttpSettingsUtil.INITIAL_WINDOW_SIZE)) {
           int initialWindowSize = OkHttpSettingsUtil.get(
               settings, OkHttpSettingsUtil.INITIAL_WINDOW_SIZE);
-          outboundFlow.initialOutboundWindowSize(initialWindowSize);
+          outboundWindowSizeIncreased = outboundFlow.initialOutboundWindowSize(initialWindowSize);
         }
         if (firstSettings) {
           listener.transportReady();
           firstSettings = false;
         }
-        startPendingStreams();
       }
 
       frameWriter.ackSettings(settings);
+
+      // Write pending bytes / streams after ack settings.
```

Comment:
I think there needs to be a stronger comment somewhere saying that we _must_ not write any frames until after the `ackSettings()`.